internal/cleanup/cleanup_job.go

```diff
@@ -0,0 +1,74 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package cleanup
+
+import (
+	"context"
+	"time"
+
+	"github.com/hashicorp/boundary/internal/db"
+	"github.com/hashicorp/boundary/internal/errors"
+	"github.com/hashicorp/boundary/internal/scheduler"
+)
+
+type cleanupJob struct {
+	r db.Reader
+	w db.Writer
+}
+
+func newCleanupJob(ctx context.Context, r db.Reader, w db.Writer) (*cleanupJob, error) {
+	const op = "cleanupJob.newCleanupJob"
+	switch {
+	case r == nil:
+		return nil, errors.New(ctx, errors.InvalidParameter, op, "missing db.Reader")
+	case w == nil:
+		return nil, errors.New(ctx, errors.InvalidParameter, op, "missing db.Writer")
+	}
+
+	return &cleanupJob{
+		r: r,
+		w: w,
+	}, nil
+}
+
+// Status reports the job’s current status.
+func (c *cleanupJob) Status() scheduler.JobStatus {
+	return scheduler.JobStatus{}
+}
+
+// Run performs the required work depending on the implementation.
+// The context is used to notify the job that it should exit early.
+func (c *cleanupJob) Run(ctx context.Context) error {
+	const op = "cleanup.(cleanupJob).Run"
+	if _, err := c.w.DoTx(ctx, db.StdRetryCnt, db.ExpBackoff{},
+		func(_ db.Reader, w db.Writer) error {
+			_, err := w.Exec(ctx, `select * from cleanup_deleted_tables()`, nil)
```

Comment:
According to https://www.postgresql.org/docs/current/sql-syntax-calling-funcs.html#SQL-SYNTAX-CALLING-FUNCS-POSITIONAL, we should be able to make this just
```suggestion
			_, err := w.Exec(ctx, `select cleanup_deleted_tables()`, nil)
```